monitor/liquidator.go

```diff
@@ -19,43 +19,64 @@ package monitor
 import (
 	"fmt"
 	"strings"
+	"sync"
+	"sync/atomic"
 	"time"
 
 	"github.com/coinbase-samples/prime-liquidator-go/config"
 	"github.com/coinbase-samples/prime-liquidator-go/monitor/caller"
-	"github.com/coinbase-samples/prime-liquidator-go/prime"
+	"github.com/coinbase-samples/prime-liquidator-go/util"
+	prime "github.com/coinbase-samples/prime-sdk-go"
 	"github.com/shopspring/decimal"
-	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
 )
 
 type Liquidator struct {
-	config         config.AppConfig
+	config         *config.AppConfig
 	convertSymbols caller.ConvertSymbols
-	balances       []*prime.AssetBalances
+	balances       []*prime.Balance
 	products       caller.ProductLookup
 	wallets        caller.WalletLookup
 	call           caller.Caller
+	stopWaitGroup  sync.WaitGroup
+	running        atomic.Bool
 }
 
-// RunLiquidator continuously assets and changes them into
+// StartLiquidator continuously assets and changes them into
 // fiat.
-func RunLiquidator(config config.AppConfig) {
+func StartLiquidator(config *config.AppConfig) (*Liquidator, error) {
 
 	l := newLiquidator(config)
 
-	l.monitor()
+	l.stopWaitGroup.Add(1)
+
+	l.running.Store(true)
+
+	go l.monitor()
+
```

Comment:
should there be logging at this higher level to describe overall state of the liquidator (starting, running, stopped)?

Reply:
There is overall logging in cmd/server/main.go, which I see as a better place to log